[Feature][Connector-V2] add tdengine source

lhyundeadsoul opened the pull request:
<!--

Thank you for contributing to SeaTunnel! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

## Contribution Checklist

  - Make sure that the pull request corresponds to a [GITHUB issue](https://github.com/apache/incubator-seatunnel/issues).

  - Name the pull request in the form "[Feature] [component] Title of the pull request", where *Feature* can be replaced by `Hotfix`, `Bug`, etc.

  - Minor fixes should be named following this pattern: `[hotfix] [docs] Fix typo in README.md doc`.

-->

## Purpose of this pull request

refer to https://github.com/apache/incubator-seatunnel/issues/2671
<!-- Describe the purpose of this pull request. For example: This pull request adds checkstyle plugin.-->

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

Commit:
add tdengine source

Commit:
add tdengine source

Commit:
add tdengine source

Commit:
remove ut

Commit:
remove ut

hailin0 commented:
can you add e2e & docs ?

reference:
https://github.com/apache/incubator-seatunnel/tree/dev/seatunnel-e2e/seatunnel-flink-connector-v2-e2e
https://github.com/apache/incubator-seatunnel/tree/dev/seatunnel-e2e/seatunnel-spark-connector-v2-e2e
https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2/source

Commit:
add doc & fix type convert error

Commit:
remove magic number

Hisoka-X committed:
Merge branch 'dev' into tdengine

songjianet committed:
[Feat][UI] Add login and logout function in the project. (#2769)

TyrantLucifer committed:
[Bug][Core] Fix the bug that can not convert array and map (#2750)

* [Bug][Core] Fix the bug that can not convert array and map

* [Hotfix][SeaTunnel-Schema] Fix the bug that can not parse decimal

* [Bug][Connector-V2] Fix the bug that can not convert map and array correctly

* [Bug][Core] Optimize code

TyrantLucifer committed:
[Bug][Connector-V2] Fix the bug that can not print SeaTunnelRow correctly (#2749)

* [Bug][Connector-V2] Fix the bug that can not print SeaTunnelRow correctly

* [Bug][Connector-V2] Fix test case

songjianet committed:
[Feat][UI] Update login function in the project. (#2773)

songjianet committed:
[Feat][UI] Add user list function in the project. (#2780)

laglangyue committed:
[connector-v1][doc]Fix inconsistency between document and code logic (#2776)

* fix connector-v1 jdbc doc exception

Co-authored-by: laglangyue <laglanyue@foxmail.com>

Hisoka-X committed:
[Connector-V2] [Kafka] Fix Kafka Streaming problem (#2759)

* [Connector-V2] [Kafka] Fix Kafka Streaming problem

TyrantLucifer committed:
[Bug][Connector-V2] Fix error option (#2775)

ic4y committed:
fix spark execute exception is not thrown (#2791)

TyrantLucifer committed:
[Bug][Connector-V2] Fix hive source text table name (#2797)

songjianet committed:
[Feat][UI] Add user delete, disable and enable function in the project. (#2795)

ashulin committed:
[Improve][e2e] Improve end-to-end testing of connector-v2 in a multi-engine environment (#2762)

songjianet committed:
[Feat][UI] Add user add and update function in the project. (#2810)

* [Feat][UI] Add user add and update function in the project.

* [Feat][UI] Add user add and update function in the project.

EricJoy2048 committed:
[Fix] Fix hive source doc (#2804)

* Fix hive source doc

* add key features in hive,local,hdfs,oss file source connector

TyrantLucifer committed:
[Improve][Connector-V2] Imporve orc read strategy (#2747)

* [Improve][Connector-V2] Imporve read orc

CalvinKirs committed:
[License]Update binary license (#2798)

* [License]Update binary license
Connector and Web are no longer in the core binary package, so their licenses are no longer managed

* open check script

hailin0 committed:
[Improve][Connector-V2] Improve mongodb connector (#2778)

* Fix schema projection
* Fix datatype conversion between MongoDB and SeaTunnel
* Support use `SimpleTextSchema` read & write

ic4y committed:
fix XAConnection being wrongly submitted (#2805)

hailin0 committed:
[Bug][seatunnel-translation-spark] Fix compile version exception (#2827)

liugddx committed:
[Doc][Connector-V2] add Console sink doc (#2822)

* [Doc][connector] add Console sink doc

* [Improve][DOC] A little tinkering

* [Improve][DOC] A little tinkering

TyrantLucifer committed:
[Bug] [Connector-V2] Fix hive source connector parallelism not work (#2823)

* [Bug][Connector-V2] Hive Source Connector parallelism not work

* [Bug][Connector-V2] Fix code style

* [Bug][Connector-V2] Fix assigned splits
